### PR TITLE
Update docs for Java

### PR DESCRIPTION
**Pull request by mdpiper**
This PR updates the BMI documentation to include Java as a supported language.

### CODE REVIEW

File: docs/source/bmi.best_practices.rst [hunk 1]
```diff
@@ -30,10 +30,10 @@ here are some tips to help when writing a BMI for a model.
...
   match input and output variables between models.
 
-* Don't change the variable names you currently use within your model
-  to :term:`Standard Names`. Standard Names are too long and
-  cumbersome to be used within a model code. Instead, you find a
-  `matching`_ Standard Name for each variable in your model and then
+* Don't change the variable names for exchange items 
+  you currently use within your model
+  to :term:`Standard Names`. Instead, find a
+  `matching`_ Standard Name for each of these variable and then
```
Comment (mcflugen) [inline docs/source/bmi.best_practices.rst:36]: *Standard Name for each variable and then*

File: README.rst [hunk 2]
```diff
@@ -75,21 +74,14 @@ are listed in the table below.
...
    Fortran   `bmi-fortran`_  `bmi-example-fortran`_
+   Java      `bmi-java`_     `bmi-example-java`_
    Python    `bmi-python`_   `bmi-example-python`_
    ========  ==============  ======================
 
 Detailed instructions for building the specifications and examples
 are given at each link above.
-The specifications can also be installed through `conda`.
 
-.. code-block:: bash
```
Comment (mcflugen) [inline README.rst:85]: Why did you remove the install instructions?

**Review by mcflugen — CHANGES_REQUESTED** [review]
Looks great. Thanks!

File: README.rst (hunk 2 above)
Comment (mdpiper) [reply README.rst:85]: More detailed explanations are given with each of the specs and examples, so this is redundant. And the Java spec and example uses Maven.

As a compromise, how about this:
> Alternatively, the specifications can be installed through conda (C, C++, Fortran, Python) or Maven (for Java). See the links above for details.

Comment (mcflugen) [reply README.rst:85]: 👍

**Review by mcflugen — APPROVED** [review]
👍
